File: app/src/main/java/com/molihua/hlbmerge/utils/FragmentTools.java
package com.molihua.hlbmerge.utils;

import androidx.fragment.app.FragmentTransaction;

import com.molihua.hlbmerge.R;
import com.molihua.hlbmerge.fragments.MoreChooseFragment;

public class FragmentTools {
    private static MoreChooseFragment moreChooseFragment;
    //显示多选按钮
    public static MoreChooseFragment showMoreChooseBtn(FragmentTransaction fragmentTransaction){

        if (moreChooseFragment ==null){
            moreChooseFragment = new MoreChooseFragment();//实例化fragment
            fragmentTransaction.add(R.id.show_hidden_bar, moreChooseFragment);//添加fragment
        }

        fragmentTransaction.show(moreChooseFragment);//显示fragment


        fragmentTransaction.commitAllowingStateLoss();//提交事务
        return moreChooseFragment;

    }
    //隐藏多选按钮
    public static MoreChooseFragment HiddenMoreChooseBtn(FragmentTransaction fragmentTransaction){

        if (moreChooseFragment ==null){
            moreChooseFragment = new MoreChooseFragment();//实例化fragment
            fragmentTransaction.add(R.id.show_hidden_bar, moreChooseFragment);//添加fragment
        }

        fragmentTransaction.hide(moreChooseFragment);//隐藏fragment

        fragmentTransaction.commitAllowingStateLoss();//提交事务
        return moreChooseFragment;

    }
}

File: app/src/main/java/com/molihua/hlbmerge/utils/FragmentTools.kt
package com.molihua.hlbmerge.utils

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.molihuan.pathselector.utils.Mtools

object FragmentTools {
    /**
     * 显示 隐藏 fragment
     *
     * @param fragmentManager Fragment管理者
     * @param frameLayoutID   添加的地方frameLayout
     * @param fragment        添加的布局Fragment
     * @param tag             tag
     * @param isShow          是否显示
     * @return
     */
    @JvmStatic
    fun fragmentShowHide(
        fragmentManager: FragmentManager,
        frameLayoutID: Int,
        fragment: Fragment?,
        tag: String?,
        isShow: Boolean
    ): Fragment? {
        // Fragment获取事务
        var fragmentTransaction = fragmentManager.beginTransaction()

        if (fragment == null) {
            Mtools.log("fragment is null and Unable to add")
            return fragment
        }
        //判断是否已经被添加过了
        if (!fragment.isAdded()) {
            try {
                fragmentTransaction = fragmentTransaction.add(frameLayoutID, fragment, tag)
            } catch (e: Exception) {
                Mtools.log("frameLayoutID may not exist and cannot be added")
                e.printStackTrace()
            }
        }

        if (isShow) {
            //显示fragment
            fragmentTransaction.show(fragment)
        } else {
            //隐藏fragment
            fragmentTransaction.hide(fragment)
        }

        //提交事务
        fragmentTransaction.commitAllowingStateLoss()
        return fragment
    }

    @JvmStatic
    fun fragmentReplace(
        fragmentManager: FragmentManager,
        frameLayoutID: Int,
        fragment: Fragment?,
        tag: String?
    ): Fragment? {
        // Fragment获取事务
        var fragmentTransaction = fragmentManager.beginTransaction()

        if (fragment == null) {
            Mtools.log("fragment is null and Unable to replace")
            return fragment
        }

        try {
            fragmentTransaction = fragmentTransaction.replace(frameLayoutID, fragment, tag)
        } catch (e: Exception) {
            Mtools.log("frameLayoutID may not exist and cannot be replace")
            e.printStackTrace()
        }


        //提交事务
        fragmentTransaction.commitAllowingStateLoss()
        return fragment
    }
}